Added support for running manual and automated tests written in TS.

martnpaneq opened the pull request:
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other (tests): Support for running manual and automated tests written in TypeScript. Remember that test runners do not validate types as we use `esbuild` which ignores it. Closes ckeditor/ckeditor5#14170, ckeditor/ckeditor5#14171.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

psmyrek reviewed (commented):
Manual and automated tests in TypeScript work fine 🎉 

Remarks:
1. I found that the build directory for manual tests contains `.ts` files, which are not needed there (they are not imported by `.html` files):

   <img src="https://github.com/ckeditor/ckeditor5-dev/assets/56868128/f68e35d2-9151-42dc-8a22-61340d5c8d46" width="250px">

   Would it be possible to not copy TypeScript files?

2. In `lib/tasks/runautomatedtests.js` there is `IGNORE_GLOBS` that also should skip TypeScript files.

3. I think that a few sentences in the CKE5 docs (contributing section) about the fact that manual and automated tests can be written now in both JavaScript and TypeScript would be useful.

martnpaneq commented:
> Manual and automated tests in TypeScript work fine 🎉
> 
> Remarks:
> 
> 1. I found that the build directory for manual tests contains `.ts` files, which are not needed there (they are not imported by `.html` files):
>    <img alt="" width="250px" src="https://user-images.githubusercontent.com/56868128/241868577-f68e35d2-9151-42dc-8a22-61340d5c8d46.png">
>    Would it be possible to not copy TypeScript files?
> 2. In `lib/tasks/runautomatedtests.js` there is `IGNORE_GLOBS` that also should skip TypeScript files.
> 3. I think that a few sentences in the CKE5 docs (contributing section) about the fact that manual and automated tests can be written now in both JavaScript and TypeScript would be useful.

@psmyrek 
1. Fixed.
2. Good catch. I did not test running all tests and without this change the tests would fail.
3. I added a PR for `ckeditor5` with some info about TS support in the `testing-environment` section. I'm not sure if it's the best place, so if you have a better idea please let me know https://github.com/ckeditor/ckeditor5/pull/14285

psmyrek reviewed (approved):
LGTM 🎉

pomek reviewed (approved):
LGTM.

It works as expected, tests are compiled, but we lost type checking due to esbuild. It's fine when prototyping things, tho. I feel that at some point, we will bring back `ts-loader` behind some flag.
